msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/MessageForumStatisticsBean.java

```diff
@@ -499,6 +532,25 @@ public List getAllUserStatistics(){
 		//clear any gradebook info:
 		resetGradebookVariables();
 
+		// Determine whether the site is a pure anonymous scenario (Ie. all topics in the site are anonymous)
+		String siteId = ToolManager.getCurrentPlacement().getContext();
+		List<Topic> allTopics = forumManager.getTopicsInSite(siteId);
```

Comment:
I'm just spit-balling here, but could we leverage the database to return only non-anonymous topics in a new method? That would eliminate having to loop over all topics to find a non-anonymous one. You simply call the method to get all non-anon, size/empty check it and be done with it. Just an idea.